### PR TITLE
Adjust to dev tibble (tibble > 1.4.2)

### PR DESCRIPTION
**Pull request by jennybc**
I'm helping @krlmlr to assess the potential impact of changes in dev tibble. I selected this package as one of the experiments. This PR passes tests for me with dev tibble (https://github.com/tidyverse/tibble/commit/6d5368082dfce1e452c1ddd01f3a783c44d12d46) but will not pass with CRAN tibble w/o further tweaks.

I actually think it would be easy for you to adjust so that DiagrammeR works with CRAN or dev tibble. I comment inline.

### CODE REVIEW

File: R/add_nodes_from_df_cols.R [hunk 1]
```diff
@@ -138,7 +138,7 @@ add_nodes_from_df_cols <- function(graph,
           trimws() %>%
           stringr::str_split(" ") %>%
           purrr::flatten_chr() %>%
-          dplyr::as_tibble() %>%
+          dplyr::as_tibble(.name_repair = "unique") %>%
```
Comment (jennybc) [inline R/add_nodes_from_df_cols.R]: I can't really tell what's going on here, but I suspect you'd be able to rewrite this pipeline to eliminate the need to mince around the names of this intermediate object.

File: R/render_graph.R [hunk 2]
```diff
@@ -167,8 +167,7 @@ render_graph <- function(graph,
         "fillcolor" %in% colnames(graph$nodes_df)) {
 
       graph$nodes_df$fontcolor <-
-        graph$nodes_df$fillcolor %>%
-        dplyr::as_data_frame() %>%
+        tibble::tibble(value = graph$nodes_df$fillcolor) %>%
```
Comment (jennybc) [inline R/render_graph.R]: This modification seems like a neutral-to-good idea, regardless.

File: R/trav_both.R [hunk 3]
```diff
@@ -282,7 +282,7 @@ trav_both <- function(graph,
     as.integer()
 
   valid_nodes <-
-    dplyr::as_tibble(valid_nodes) %>%
+    dplyr::tibble(value = valid_nodes) %>%
```
Comment (jennybc) [inline R/trav_both.R]: This modification seems like a neutral-to-good idea, regardless.

**Comment by rich-iannone** [comment]
Jenny, thanks for this! I'll gladly accept these PRs. I need to get this package to CRAN soon, I think.

**Comment by jennybc** [comment]
Wait -- the current version here requires dev tibble, which you probably don't want!

See my comment on the first diff.

**Comment by rich-iannone** [comment]
Ah, of course... sorry about that! I reverted and will go through a function rewrite to see what is up. I remember not being very satisfied with `add_nodes_from_df_cols()`, even after a few passes with it.

**Comment by rich-iannone** [comment]
Will do. Thanks again!

**Comment by jennybc** [comment]
Does it make any sense to you that my PR passed tests on travis with r-release and, allegedly, CRAN tibble?

**Comment by jennybc** [comment]
Context: I have seen some other passing results on travis that surprise me.

**Comment by jennybc** [comment]
OK I understand why this worked now, against both CRAN and dev tibble. It's a bit of a happy accident. But you could take the diffs in the PR if you are pressed for time, as part of a soon-ish release 🙏

However, I still think it would be better for you (= someone who understands the code) to consider a refactoring of the first diff to avoid passing through the state of tibble with no names.

Here are some download stats for tibble revdeps affected by the name changes:

```
#> # A tibble: 97 x 2
#>    package          n
#>    <chr>        <dbl>
#>  1 readxl       78539
#>  2 xgboost       9175
#>  3 DiagrammeR    5378
#>  4 prophet       2782
#>  5 bsts          2751
#>  6 data.tree     2156
#>  7 feather       2042
```

I'm on top of readxl. The xgboost problem is actually the DiagrammeR problem. So making readxl and DiagrammeR compatible with CRAN + dev tibble covers an awful lots of "download ground".

**Comment by rich-iannone** [comment]
Jenny, for the first time ever I reverted something I reverted before! So the original PR’s commits are back in (thanks again)! I will get to that refactor, and yeah, hard pressed for time these days because of work on that secret package. Do you figure that a release of DiagrammeR would be a good idea now?

**Comment by rich-iannone** [comment]
I was planning on it anyway and this is a good incentive. Will try to do this in the next 1-2 days.

**Comment by jennybc** [comment]
1-2 days is definitely faster than we need, FYI. We're still assessing what can go in the release (hence these exploratory revdep checks).

**Comment by jennybc** [comment]
You might want to get ready for release ... then sit on it a little bit, in case anything else comes up.

File: R/render_graph.R (hunk 2 above)
Comment (jennybc) [reply R/render_graph.R]: I think a solution that works with 1.4.2 is better, since the sequencing of the releases becomes simpler.

**Pull request merged**